Bug Report: Incorrect logic defined in the AzAutoGen.py script

**Describe the bug**
The following lines in the script are suspicious since the for loop just executes a single line repeatedly without any difference. It also causes a potential issue where `dataInputFiles` could be referenced before defined when `fnmatch.filter(sourceFiles, inputFiles)` returns empty.

https://github.com/o3de/o3de/blob/development/cmake/AzAutoGen.py#L290-L292

**Steps to reproduce**
1. Sync branch: https://github.com/aws-lumberyard-dev/o3de/tree/junbo_jiawei_LBC
2. Configure the [MultiplayerPool](https://github.com/aws-lumberyard-dev/o3de/tree/junbo_jiawei_LBC/MultiplayerPool) created by the branch using CMake command: `cmake -B build\windows -S . -G "Visual Studio 16 2019" -DLY_3RDPARTY_PATH="%LY_3RDPARTY_PATH%" -DLY_PROJECTS=MultiplayerPool`

**Expected behavior**
The project can be configured successfully.

**Actual behavior**
CMake configuration command threw an error complaining about the undefined `dataInputFiles` variable:
```
-- Using Python 3.7.10 at D:/github/o3de/python/runtime/python-3.7.10-rev2-windows/python/python.exe
*.AutoComponent.xml,AutoComponentTypes_Header.jinja,$path/AutoComponentTypes.h : error Processing expansion rule
An unexpected error occurred, please report the error you encountered and include your build output
Traceback (most recent call last):
  File "D:/github/o3de/cmake/AzAutoGen.py", line 372, in <module>
    autoGenResult = ExecuteExpansionRules(cacheDir, outputDir, projectDir, inputFiles, expansionRules, dryrun, verbose, dataInputSet, outputFiles)
  File "D:/github/o3de/cmake/AzAutoGen.py", line 326, in ExecuteExpansionRules
    ProcessExpansionRule(sourceFiles, templateFiles, templateCache, outputDir, projectDir, expansionRule, dryrun, verbose, dataInputSet, outputFiles)
  File "D:/github/o3de/cmake/AzAutoGen.py", line 292, in ProcessExpansionRule
    outputFileAbsolute = outputFile.replace("$path", ComputeOutputPath(dataInputFiles, projectDir, outputDir))
UnboundLocalError: local variable 'dataInputFiles' referenced before assignment
CMake Error at cmake/LyAutoGen.cmake:34 (add_custom_command):
  add_custom_command Wrong syntax.  A TARGET or OUTPUT must be specified.
Call Stack (most recent call first):
  cmake/LYWrappers.cmake:381 (ly_add_autogen)
  MultiplayerPool/Code/CMakeLists.txt:24 (ly_add_target)
```

**Found in Branch**
Development, 
https://github.com/aws-lumberyard-dev/o3de/tree/junbo_jiawei_LBC

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows

**Additional context**
Add any other context about the problem here.

--- a/cmake/AzAutoGen.py
+++ b/cmake/AzAutoGen.py
@@ -287,8 +287,7 @@ def ProcessExpansionRule(sourceFiles, templateFiles, templateCache, outputDir, p
             else:
                 # Process all matches in one batch
                 # Due to the lack of wildcards in the output file, we've determined we'll glob all matching input files into the template conversion 
-                for filename in fnmatch.filter(sourceFiles, inputFiles):
-                    dataInputFiles = [os.path.abspath(file) for file in fnmatch.filter(sourceFiles, inputFiles)]
+                dataInputFiles = [os.path.abspath(file) for file in fnmatch.filter(sourceFiles, inputFiles)]
                 outputFileAbsolute = outputFile.replace("$path", ComputeOutputPath(dataInputFiles, projectDir, outputDir))
                 outputFileAbsolute = SanitizePath(outputFileAbsolute)
                 ProcessTemplateConversion(dataInputSet, dataInputFiles, templateFile, outputFileAbsolute, templateCache, dryrun, verbose)

